src/SendGrid/Helpers/Mail/Utilities/EmailAddressComparer.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace SendGrid.Helpers.Mail.Utilites
+{
+    /// <summary>
+    /// EmailAddress comparer to if check two objects are equal and generate Hashcode for an object.
+    /// </summary>
+    internal class EmailAddressComparer : IEqualityComparer<EmailAddress>
+    {
+        /// <inheritdoc/>
+        public bool Equals(EmailAddress x, EmailAddress y)
+        {
+            if (ReferenceEquals(x, y))
+            {
+                return true;
+            }
+
+            if (ReferenceEquals(x, null) || ReferenceEquals(y, null))
+            {
+                return false;
+            }
+
+            return !string.IsNullOrWhiteSpace(x.Email) && StringComparer.OrdinalIgnoreCase.Equals(x.Email, y.Email);
```

Comment:
The Sendgrid API also fails on things like `somebody@mymail.com' and 'Somebody <somebody@mymail.com>`. Your comparison here isn't necessarily good enough,